packages/clerk-js/src/core/clerk.ts

```diff
@@ -741,8 +738,13 @@ export class Clerk implements ClerkInterface {
       newSession = this.#getSessionFromClient(newSession?.id);
     }
 
-    await this.#authService?.setAuthCookiesFromSession(newSession);
-
+    // Sync __session and __client_uat to cookies using events.TokenUpdate dispatched event
+    // only for newSession is null since the getToken will not be executed. Since getToken
+    // triggers internally a events.TokenUpdate there is no need to trigger it when the newSession exists.
+    const token = await newSession?.getToken();
+    if (!token) {
+      eventBus.dispatch(events.TokenUpdate, { token: null });
+    }
```

Comment:
💭  given the comment, this feels like it should live in `getToken()`, if we're already dispatching the event when the token updates anyways. What do you think?

Reply:
This change cannot live in the `getToken()` since this covers the case of the `newSession` being null. In that case the `getToken()` will not be executed but the code will work due to `?.` operator.